libevmasm/PeepholeOptimiser.cpp

```diff
@@ -487,6 +498,99 @@ struct UnreachableCode
 	}
 };
 
+struct DeduplicateNextTagSize3 : SimplePeepholeOptimizerMethod<DeduplicateNextTagSize3>
+{
+	static bool applySimple(
+		AssemblyItem const& _itemA,
+		AssemblyItem const& _itemB,
+		AssemblyItem const& _breakingItem,
+		AssemblyItem const& _tag,
+		AssemblyItem const& _itemC,
+		AssemblyItem const& _itemD,
+		AssemblyItem const& _breakingItem2,
+		std::back_insert_iterator<AssemblyItems> _out
+	)
+	{
+		if (
+			_itemA == _itemC &&
+			_itemB == _itemD &&
+			_breakingItem == _breakingItem2 &&
+			_tag.type() == Tag &&
+			(
+				_breakingItem == Instruction::JUMP ||
```

Comment:
Which cases do we loose, if we just *not* apply this for ``JUMP``s at all here? (i.e. remove that part)?
If that's too much, then further which cases do we loose if we check for jumps, if the jump type (``getJumpType``) is the same and only do it in that case?

Reply:
I also actually don't quite remember what made us add these peephole optimizations here in the first place :-)... we did get some weird ``revert(0x00,0x00) tag_2: revert(0x00,0x00)`` patterns IIRC... I wonder where they were even coming from, though... but if it was only those, then removing the jump case would be fine for getting the rest finished...